newton/utils/import_usd.py

```diff
@@ -319,16 +319,19 @@ def parse_body(rigid_body_desc, prim, incoming_xform=None):
         if incoming_xform is not None:
             origin = wp.mul(incoming_xform, origin)
         path = str(prim.GetPath())
+        name = str(prim.GetName())
 
         body_armature = parse_float_with_fallback(
             (prim, physics_scene_prim), "warp:armature", builder.default_body_armature
         )
 
         b = builder.add_body(
             xform=origin,
-            key=path,
+            key=name,
             armature=body_armature,
         )
+        builder.add_joint_free(b)
+        builder.joint_q[-7:] = list(origin.p) + list(origin.q)
```

Comment:
These lines are needed to because I need to add a free joint and to shift the joint transform to match the body's transform. Line 333-334 ensures that an example of dropping a bottle to a crate puts the bottle in the correct position in `sim_usd`.

Reply:
I think you don't need to assign `joint_q` here, when adding a free joint we will take the child body's xform to initialize its `joint_q` values.